src/Compilers/Core/Portable/Symbols/Attributes/CommonParameterWellKnownAttributeData.cs

```diff
@@ -109,5 +109,37 @@ public bool HasIUnknownConstantAttribute
             }
         }
         #endregion
+
+        private bool _hasNotNullWhenFalseAttribute;
+        public bool HasNotNullWhenFalseAttribute
+        {
+            get
+            {
+                VerifySealed(expected: true);
+                return _hasNotNullWhenFalseAttribute;
+            }
+            set
+            {
+                VerifySealed(expected: false);
+                _hasNotNullWhenFalseAttribute = value;
+                SetDataStored();
+            }
+        }
+
+        private bool _hasEnsuresNotNullAttribute;
+        public bool HasEnsuresNotNullAttribute
```

Comment:
>HasEnsuresNotNullAttribute [](start = 20, length = 26)

I assume VB doesn't take advantage of this member, please move it to C# specific type.